File: leet_code/two-sum.go
package main

import "fmt"

func twoSum(nums []int, target int) []int {
	// var diffs = make([]int, len(nums))
	var diffs = make(map[int]int)
	for index, value := range nums {
		var diff int = target - value
		diffs[diff] = index
	}

	result := []int{0, 0}
	for index, value := range nums {
		diff_index, ok := diffs[value]
		if !ok {
			continue
		}
		if index == diff_index {
			continue
		}
		result[0] = index
		result[1] = diff_index

	}

	return result
}

func main() {
	var nums = []int{2, 7, 11, 15}
	var target int = 9

	var result = twoSum(nums, target)
	fmt.Println(result)
}

File: leet_code/two-sum.py
from typing import List, Tuple


class Solution:
    def twoSum(self, nums: List[int], target: int) -> Tuple[int, int]:

        diffs = {}
        for i, value in enumerate(nums):
            try:
                if diffs[target-value] is not None:
                    return diffs[target-value], i
            except KeyError:
                pass
            diffs[value] = i

        a = 1



if __name__ == "__main__":
    solution = Solution()

    res = solution.twoSum([3,3], 6)  #
    print(res)